### PR TITLE
[dev] PBI AB#55858: Action Bar Mobile Search Auto Focus

### PR DESCRIPTION
**Pull request by cbrewer1689**

### CODE REVIEW

File: src/surfaces/actionBar/actionBar.jsx [hunk 1]
```diff
@@ -180,6 +189,25 @@ class ActionBar extends React.PureComponent {
...
+        } = this.props;
+
+        let hasEventBeenCalled = false;
+
+        forEach(columns, (c) => {
+            if (c.list) {
+                forEach(c.list, (l) => {
+                    if (!hasEventBeenCalled && l.iconSearch?.autoFocus) {
+                        this.onMobileSearchIconToggle();
+                        hasEventBeenCalled = true;
```
Comment (groberts314) [inline src/surfaces/actionBar/actionBar.jsx:204]: Should we add `break;` here as well to short-circuit any remaining loop iterations?

File: src/surfaces/actionBar/actionBar.jsx [hunk 2]
```diff
@@ -84,7 +89,7 @@ const defaultProps = {
 };
 
 const HAS_DRAWER_ACTION_BAR_CLASS_NAME = 'drawer-has_action_bar';
-const HAS_PAGE_ACTION_BAR_CLASS_NAME = 'page-has_action_bar';
+const HAS_PAGE_ACTION_BAR_CLASS_NAME = 'page-has_action\_bar';
```
Comment (groberts314) [inline src/surfaces/actionBar/actionBar.jsx:92]: Is that back slash intended there?

**Review by groberts314 — CHANGES_REQUESTED** [review]
A couple of small things

File: src/surfaces/actionBar/actionBar.jsx (hunk 2 above)
Comment (cbrewer1689) [reply src/surfaces/actionBar/actionBar.jsx:92]: Nope, that is not intended. Thanks for the catch.

File: src/surfaces/actionBar/actionBar.jsx (hunk 1 above)
Comment (cbrewer1689) [reply src/surfaces/actionBar/actionBar.jsx:204]: You can use `break;` in a forEach? Here's what I get from my editor...

image

Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx:204]: Stinkin' JavaScript!  You can totally do that in C#!
Ah well ... don't mind me!  Nothing to see here! =)

File: src/surfaces/actionBar/actionBar.jsx (hunk 2 above)
Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx]: Lookin' better.  Resolving.

**Review by groberts314 — APPROVED** [review]

**Review by War777 — APPROVED** [review]

File: src/surfaces/actionBar/actionBar.jsx (hunk 1 above)
Comment (cbrewer1689) [reply src/surfaces/actionBar/actionBar.jsx:204]: You know what is funny, I thought while making this change, "I wish you could break out of these loops easily, but I cannot." This is probably obvious as to why i implemented the `hasEventBeenCalled` boolean. Then you commented, about `break;` and I thought, "oh, man, you can, that's cool." Then JS gave me the old Jurassic Park computer password warning.

Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx:204]: https://www.youtube.com/watch?v=rKX4KmtSBk8

Comment (groberts314) [reply src/surfaces/actionBar/actionBar.jsx:204]: Classic! =)

**Pull request merged**